app/src/ui/repository-settings/repository-settings.tsx

```diff
@@ -286,9 +286,10 @@ export class RepositorySettings extends React.Component<
   private onSubmit = async () => {
     this.setState({ disabled: true, errors: undefined })
     const errors = new Array<JSX.Element | string>()
+    const modifiedUrl = this.state.remote?.url.trim()
```

Comment:
This should be inside of the first `if`, so you don't need the `?`. Then, of course, you should use `modifiedUrl` (I'd name it `trimmedURL`) as parameter to `setRemoteURL` instead of `this.state.remote.url`.

Reply:
done